usd-rs/src/pxr/usd/prim.rs

```diff
@@ -77,6 +108,37 @@ impl Prim {
             })
         }
     }
+    
+    // this was causing segfault, hence the dirty implementation below. 
+/*     pub fn get_attributes(&self) -> Vec<Attribute> {
```

Comment:
This was seg-faulting because rust Vec type is a different type to std::vector.
You were right in copying into a Vec below. It might be possible to write this in less code though if you fills the Vec from a slice.